app/src/main/java/org/matrix/matrixandroidsdk/activity/HomeActivity.java

```diff
@@ -257,7 +259,13 @@ public boolean onChildClick(ExpandableListView parent, View v,
                     roomId = mAdapter.getPublicRoomAt(childPosition).roomId;
                 }
 
-                CommonActivityUtils.goToRoomPage(roomId, HomeActivity.this);
+                if (Intent.ACTION_SEND.equals(getIntent().getAction()) && getIntent().getType() != null) {
```

Comment:
Parsing the ``Intent`` here seems wrong to me. From this, it looks like if I were to share something with Matrix, I would be dumped into the Home Screen without any indication that I'm sharing something, then only when I click on a room will it auto-populate the text field with the text I planned on sharing. 

It may be worth having another ``Activity`` manage external ``Intents`` like this, so you can present a screen that much more clearly says "Select a room to share [text/image/whatever]". This would have the added benefit of keeping ``HomeActivity``'s complexity down, as it would be handling less things.